### ISSUE TITLE
PeriodSerializer not working with midnight DATE-TIME.

### ISSUE THREAD

**Issue by minichma** [rianjs/ical.net#564]
PeriodSerializer doesn't deserialize a period's duration correctly if the start IDateTime's time is set to midnight.

E.g. the following period string should be serialized to a period with a duration of 6H but actually has 1D:
`20180219T000000Z/PT6H`

Can be reproduced via the following test case:

```CS
[Test]
public void TestPeriods()
{
    var serializer = new PeriodSerializer();
    var res = serializer.Deserialize(new System.IO.StringReader("20180219T000000Z/PT6H")) as Period;

    Assert.AreEqual(6.0, res.Duration.Hours);
    Assert.IsTrue(res.StartTime.HasTime);
 }
```

The problem boils down to `Period` [automatically assuming a duration of 1D if the time component is zero](https://github.com/rianjs/ical.net/blob/1ad17e51a0e71b30aabf88d95c16066f4db6e08b/src/Ical.Net/DataTypes/Period.cs#L142).

Also see [similar code in CalDateTime.cs](https://github.com/ical-org/ical.net/blob/35b506c479d16102b5c22f9d210c853ed1698c8d/Ical.Net/DataTypes/CalDateTime.cs#L93) and a related issue in #602.

**Comment by axunonb** [comment]
Please double-check the new version merged with #595

**Comment by minichma** [comment]
@axunonb: Yes, the problem still exists.

**Comment by minichma** [comment]
@axunonb I think we should reopen this. Don't think I can do that myself. Started working on a PR but could take a while until I find enough time, as all these little issues are interconnected.

**Comment by minichma** [comment]
@axunonb Thanks for the invitation! Reopened the relevant issues.

**Comment by minichma** [comment]
Fixed with #638
